### PR TITLE
Search filter user preferences using database data

### PR DESCRIPTION
**Pull request by gbashour**
[Update map.tsx to use database data for filters](https://trello.com/c/LMbYr82k/69-update-maptsx-to-use-database-data-for-filters)

- Grab username from JWT when user is signed in
- Fetch user preferences from the DB for a specific user
- Added logic to use this preferences to pre-set active filters on page load
- Fixed Restaurant Details routing from search results
- Other small code fixes

Screenshot 2024-10-15 at 9 30 01 PM

### CODE REVIEW

**Comment by AdamNas123** [comment]
Changes look good! I think we also need to consider the case where the toggle in User Settings is turned off, so that the user preferences do not get loaded into the search. Currently, the preferences seem to get added to the search either way.

File: food-for-thought/app/map.tsx [hunk 1]
```diff
@@ -39,14 +45,14 @@ export type Restaurant = {
...
-  rating?: number,
-  total_ratings?: number,
-  menuId?: string,
+  rating: number,
+  total_ratings: number,
+  menuId: string,
   restaurantPhotos?: string[],
   foodPhotos?: string[],
-  hasMenu?: boolean
+  hasMenu: boolean
```
Comment (azchu) [inline food-for-thought/app/map.tsx]: Just curious, are we making these non-nullable cause they are being defined elsewhere and can't be null? Just being cautious cause we may get undefined errors later down the line

File: food-for-thought/app/map.tsx [hunk 2]
```diff
@@ -67,9 +74,45 @@ const RestaurantMap = () => {
...
+  const pic = require('../assets/images/react-logo.png'); // placeholder restaurant image
+
+  const HOST_IP = Constants.expoConfig?.extra?.HOST_IP;
+
+  const loadUser = React.useCallback(async () => {
+    const token = await AsyncStorage.getItem('token');
+    if (token) {
+        try {
+            const decodedToken: any = jwtDecode(token);
+            console.log(decodedToken.username);
```
Comment (azchu) [inline food-for-thought/app/map.tsx]: not important, but would remove stray console logs (good to keep console clean ✨)

File: food-for-thought/app/map.tsx [hunk 3]
```diff
@@ -67,9 +74,45 @@ const RestaurantMap = () => {
...
+  }, []);
+
+  React.useEffect(() => {
+    loadUser();
+  }, [loadUser]);
+
+  const fetchUserPreferences = async (username: string) => {
+    /* if they're logged in, grab the preferences associated with this username */
+    try {
+      console.log(username);
```
Comment (azchu) [inline food-for-thought/app/map.tsx]: stray console log ✨

**Review by azchu — APPROVED** [review]
LGTM, just some minor comments 🎉

**Review by abrarbarbhuyia — APPROVED** [review]
LGTM

File: food-for-thought/app/map.tsx (hunk 1 above)
Comment (gbashour) [reply food-for-thought/app/map.tsx]: I changed these to reflect the `required` fields in the Restaurant Schema

File: food-for-thought/app/map.tsx (hunk 2 above)
Comment (gbashour) [reply food-for-thought/app/map.tsx]: Absolutely agree! We should remove these before merging - there's some other stray console logs on other pages as well, will look into tidying up those too🙏

**Pull request merged**